In [272]:
#Dependencies
import pandas as pd
import csv

#load files
schools_data = "Resources/schools_complete.csv"
students_data = "Resources/students_complete.csv"

#create DataFrames
school_data = pd.read_csv(schools_data)
student_data = pd.read_csv(students_data)

In [273]:
#check data loaded correctly
school_data.head()


School ID            school_name         type  size   budget
0          0      Huang High School   Government  2917  1910635
1          1   Figueroa High School   Government  2949  1884411
2          2    Shelton High School  Independent  1761  1056600
3          3  Hernandez High School   Government  4635  3022020
4          4    Griffin High School  Independent  1468   917500

In [274]:
#check data loaded correctly
student_data.head()


Student ID     student_name gender  year        school_name  reading_score  \
0           0     Paul Bradley      M     9  Huang High School             96   
1           1     Victor Smith      M    12  Huang High School             90   
2           2  Kevin Rodriguez      M    12  Huang High School             41   
3           3    Richard Scott      M    12  Huang High School             89   
4           4       Bonnie Ray      F     9  Huang High School             87   

   maths_score  
0           94  
1           43  
2           76  
3           86  
4           69

In [275]:
#merge the 2 dataframes into one combined dataframe
school_student_complete_data = pd.merge(school_data, student_data, on=["school_name"])
school_student_complete_data.head()


School ID        school_name        type  size   budget  Student ID  \
0          0  Huang High School  Government  2917  1910635           0   
1          0  Huang High School  Government  2917  1910635           1   
2          0  Huang High School  Government  2917  1910635           2   
3          0  Huang High School  Government  2917  1910635           3   
4          0  Huang High School  Government  2917  1910635           4   

      student_name gender  year  reading_score  maths_score  
0     Paul Bradley      M     9             96           94  
1     Victor Smith      M    12             90           43  
2  Kevin Rodriguez      M    12             41           76  
3    Richard Scott      M    12             89           86  
4       Bonnie Ray      F     9             87           69

## Local Government Area Summary

In [276]:
#find the total number of unique schools
unique_schools = school_data["school_name"].count()
print(unique_schools)

15


In [277]:
#find the total number of students
total_students = school_student_complete_data["student_name"].count()
print(total_students)

39170


In [278]:
#find the total budget of all schools
total_budget = school_data["budget"].sum()
print(total_budget)

24649428


In [279]:
#find the average math scores
avg_math_scores = school_student_complete_data["maths_score"].mean()
print(avg_math_scores)

70.33819249425581


In [280]:
#find the average reading scores
avg_reading_scores = school_student_complete_data["reading_score"].mean()
print(avg_reading_scores)

69.98013786060761


In [281]:
#find the percentage of students who passed maths
math_pass = school_student_complete_data.loc[school_student_complete_data["maths_score"] >=50]
percent_math_pass = len(math_pass)/total_students * 100
print(percent_math_pass)

86.07863160582077


In [282]:
#find the percentage of students who passed reading
reading_pass = school_student_complete_data.loc[school_student_complete_data["reading_score"] >=50]
percent_reading_pass = len(reading_pass)/total_students * 100
print(percent_reading_pass)

84.42685728874139


In [283]:
#find the percentage of students who passed both maths and reading 
#pass grade is 50 or higher
math_reading_pass =school_student_complete_data[(school_student_complete_data["maths_score"] >=50) 
& (school_student_complete_data["reading_score"] >=50)] 

math_reading_pass_students = math_reading_pass["student_name"].count()

overall_pass_percent = math_reading_pass_students / total_students *100
overall_pass_percent

72.80827163645647

In [284]:
area_summary_df = pd.DataFrame([{"Total Schools": unique_schools,"Total Students": total_students,"Total Budget": total_budget,
                    "Average Maths Score": avg_math_scores,"Average Reading Score": avg_reading_scores, "% Passing Maths": percent_math_pass,
                    "% Passing Reading": percent_reading_pass, "% Overall Passing": overall_pass_percent}])

area_summary_df["Total Budget"] = area_summary_df["Total Budget"].astype(float).map("${:,.2f}".format)
area_summary_df

Total Schools  Total Students    Total Budget  Average Maths Score  \
0             15           39170  $24,649,428.00            70.338192   

   Average Reading Score  % Passing Maths  % Passing Reading  \
0              69.980138        86.078632          84.426857   

   % Overall Passing  
0          72.808272

## School Summary

In [285]:
#school type
school_type = school_data.set_index(["school_name"])["type"]
school_type

school_name
Huang High School         Government
Figueroa High School      Government
Shelton High School      Independent
Hernandez High School     Government
Griffin High School      Independent
Wilson High School       Independent
Cabrera High School      Independent
Bailey High School        Government
Holden High School       Independent
Pena High School         Independent
Wright High School       Independent
Rodriguez High School     Government
Johnson High School       Government
Ford High School          Government
Thomas High School       Independent
Name: type, dtype: object

In [286]:
#total students
school_students = school_student_complete_data["school_name"].value_counts()
school_students

Bailey High School       4976
Johnson High School      4761
Hernandez High School    4635
Rodriguez High School    3999
Figueroa High School     2949
Huang High School        2917
Ford High School         2739
Wilson High School       2283
Cabrera High School      1858
Wright High School       1800
Shelton High School      1761
Thomas High School       1635
Griffin High School      1468
Pena High School          962
Holden High School        427
Name: school_name, dtype: int64

In [287]:
#Total school budget
school_budget = school_student_complete_data.groupby(["school_name"])["budget"].mean()
school_budget

school_name
Bailey High School       3124928
Cabrera High School      1081356
Figueroa High School     1884411
Ford High School         1763916
Griffin High School       917500
Hernandez High School    3022020
Holden High School        248087
Huang High School        1910635
Johnson High School      3094650
Pena High School          585858
Rodriguez High School    2547363
Shelton High School      1056600
Thomas High School       1043130
Wilson High School       1319574
Wright High School       1049400
Name: budget, dtype: int64

In [288]:
#per student budget
per_student_budget = school_budget / school_students
per_student_budget

Bailey High School       628.0
Cabrera High School      582.0
Figueroa High School     639.0
Ford High School         644.0
Griffin High School      625.0
Hernandez High School    652.0
Holden High School       581.0
Huang High School        655.0
Johnson High School      650.0
Pena High School         609.0
Rodriguez High School    637.0
Shelton High School      600.0
Thomas High School       638.0
Wilson High School       578.0
Wright High School       583.0
dtype: float64

In [289]:
#average math score
math_scores_per_school = school_student_complete_data.groupby(["school_name"])["maths_score"].mean()
math_scores_per_school

school_name
Bailey High School       72.352894
Cabrera High School      71.657158
Figueroa High School     68.698542
Ford High School         69.091274
Griffin High School      71.788147
Hernandez High School    68.874865
Holden High School       72.583138
Huang High School        68.935207
Johnson High School      68.843100
Pena High School         72.088358
Rodriguez High School    72.047762
Shelton High School      72.034072
Thomas High School       69.581651
Wilson High School       69.170828
Wright High School       72.047222
Name: maths_score, dtype: float64

In [290]:
#average reading score
reading_scores_per_school = school_student_complete_data.groupby(["school_name"])["reading_score"].mean()
reading_scores_per_school

school_name
Bailey High School       71.008842
Cabrera High School      71.359526
Figueroa High School     69.077993
Ford High School         69.572472
Griffin High School      71.245232
Hernandez High School    69.186408
Holden High School       71.660422
Huang High School        68.910525
Johnson High School      69.039277
Pena High School         71.613306
Rodriguez High School    70.935984
Shelton High School      70.257808
Thomas High School       69.768807
Wilson High School       68.876916
Wright High School       70.969444
Name: reading_score, dtype: float64

In [291]:
#% passing maths (the percentage of students who passed maths >50)
math_pass_per_school = school_student_complete_data[(school_student_complete_data["maths_score"]>=50)]
math_pass_percent_per_school = math_pass_per_school.groupby(["school_name"])["maths_score"].count() / school_students * 100 
math_pass_percent_per_school

Bailey High School       91.639871
Cabrera High School      90.850377
Figueroa High School     81.654798
Ford High School         82.438846
Griffin High School      91.212534
Hernandez High School    80.949299
Holden High School       89.929742
Huang High School        81.693521
Johnson High School      82.062592
Pena High School         91.683992
Rodriguez High School    90.797699
Shelton High School      91.538898
Thomas High School       83.853211
Wilson High School       82.785808
Wright High School       91.777778
dtype: float64

In [292]:
#% passing reading (the percentage of students who passed reading >50)
reading_pass_per_school = school_student_complete_data[(school_student_complete_data["reading_score"]>=50)]
reading_pass_percent_per_school = reading_pass_per_school.groupby(["school_name"])["reading_score"].count() / school_students * 100 
reading_pass_percent_per_school

Bailey High School       87.379421
Cabrera High School      89.074273
Figueroa High School     82.807731
Ford High School         82.219788
Griffin High School      88.487738
Hernandez High School    81.877023
Holden High School       88.524590
Huang High School        81.453548
Johnson High School      81.978576
Pena High School         86.590437
Rodriguez High School    87.396849
Shelton High School      86.712095
Thomas High School       82.629969
Wilson High School       81.296540
Wright High School       86.666667
dtype: float64

In [293]:
#% overall passing (the percentage of students who passed maths AND reading >50)
math_reading_pass_percent_per_school =school_student_complete_data[(school_student_complete_data["maths_score"]>=50)
& (school_student_complete_data["reading_score"]>=50)]

math_reading_pass_percent_per_school =math_reading_pass_percent_per_school.groupby(["school_name"])["student_name"].count()

overall_pass_percent_per_school = math_reading_pass_percent_per_school / school_students * 100
overall_pass_percent_per_school

Bailey High School       80.084405
Cabrera High School      80.785791
Figueroa High School     67.650051
Ford High School         67.469880
Griffin High School      81.335150
Hernandez High School    66.364617
Holden High School       78.922717
Huang High School        66.712376
Johnson High School      67.191766
Pena High School         79.209979
Rodriguez High School    79.419855
Shelton High School      78.875639
Thomas High School       69.480122
Wilson High School       67.455103
Wright High School       79.722222
dtype: float64

In [294]:
per_school_summary_df = pd.DataFrame({"School Type": school_type,"Total Students": school_students,
                    "Total Budget": school_budget,"Per Student Budget": per_student_budget, "Average Maths Score": math_scores_per_school, 
                    "Average Reading Score": reading_scores_per_school, "% Passing Maths": math_pass_percent_per_school,
                    "% Passing Reading": reading_pass_percent_per_school, "% Overall Passing": overall_pass_percent_per_school})

per_school_summary_df["Total Budget"] = per_school_summary_df["Total Budget"].astype(float).map("${:,.2f}".format)
per_school_summary_df["Per Student Budget"] = per_school_summary_df["Per Student Budget"].astype(float).map("${:,.2f}".format)
per_school_summary_df                    

School Type  Total Students   Total Budget  \
Bailey High School      Government            4976  $3,124,928.00   
Cabrera High School    Independent            1858  $1,081,356.00   
Figueroa High School    Government            2949  $1,884,411.00   
Ford High School        Government            2739  $1,763,916.00   
Griffin High School    Independent            1468    $917,500.00   
Hernandez High School   Government            4635  $3,022,020.00   
Holden High School     Independent             427    $248,087.00   
Huang High School       Government            2917  $1,910,635.00   
Johnson High School     Government            4761  $3,094,650.00   
Pena High School       Independent             962    $585,858.00   
Rodriguez High School   Government            3999  $2,547,363.00   
Shelton High School    Independent            1761  $1,056,600.00   
Thomas High School     Independent            1635  $1,043,130.00   
Wilson High School     Independent            2283  $1,319,574.00   
Wright High School     Independent            1800  $1,049,400.00   

                      Per Student Budget  Average Maths Score  \
Bailey High School               $628.00            72.352894   
Cabrera High School              $582.00            71.657158   
Figueroa High School             $639.00            68.698542   
Ford High School                 $644.00            69.091274   
Griffin High School              $625.00            71.788147   
Hernandez High School            $652.00            68.874865   
Holden High School               $581.00            72.583138   
Huang High School                $655.00            68.935207   
Johnson High School              $650.00            68.843100   
Pena High School                 $609.00            72.088358   
Rodriguez High School            $637.00            72.047762   
Shelton High School              $600.00            72.034072   
Thomas High School               $638.00            69.581651   
Wilson High School               $578.00            69.170828   
Wright High School               $583.00            72.047222   

                       Average Reading Score  % Passing Maths  \
Bailey High School                 71.008842        91.639871   
Cabrera High School                71.359526        90.850377   
Figueroa High School               69.077993        81.654798   
Ford High School                   69.572472        82.438846   
Griffin High School                71.245232        91.212534   
Hernandez High School              69.186408        80.949299   
Holden High School                 71.660422        89.929742   
Huang High School                  68.910525        81.693521   
Johnson High School                69.039277        82.062592   
Pena High School                   71.613306        91.683992   
Rodriguez High School              70.935984        90.797699   
Shelton High School                70.257808        91.538898   
Thomas High School                 69.768807        83.853211   
Wilson High School                 68.876916        82.785808   
Wright High School                 70.969444        91.777778   

                       % Passing Reading  % Overall Passing  
Bailey High School             87.379421          80.084405  
Cabrera High School            89.074273          80.785791  
Figueroa High School           82.807731          67.650051  
Ford High School               82.219788          67.469880  
Griffin High School            88.487738          81.335150  
Hernandez High School          81.877023          66.364617  
Holden High School             88.524590          78.922717  
Huang High School              81.453548          66.712376  
Johnson High School            81.978576          67.191766  
Pena High School               86.590437          79.209979  
Rodriguez High School          87.396849          79.419855  
Shelton High School            86.712095          78.875639  
Thomas High School            

## Top Performing Schools (By % Overall Passing)

In [295]:
#Sort the schools by % Overall Passing in descending order and display the top 5 rows.
#Save the results in a DataFrame called "top_schools".
top_schools_df = per_school_summary_df.sort_values(["% Overall Passing"], ascending=False)
top_schools_df.head(5)

School Type  Total Students   Total Budget  \
Griffin High School    Independent            1468    $917,500.00   
Cabrera High School    Independent            1858  $1,081,356.00   
Bailey High School      Government            4976  $3,124,928.00   
Wright High School     Independent            1800  $1,049,400.00   
Rodriguez High School   Government            3999  $2,547,363.00   

                      Per Student Budget  Average Maths Score  \
Griffin High School              $625.00            71.788147   
Cabrera High School              $582.00            71.657158   
Bailey High School               $628.00            72.352894   
Wright High School               $583.00            72.047222   
Rodriguez High School            $637.00            72.047762   

                       Average Reading Score  % Passing Maths  \
Griffin High School                71.245232        91.212534   
Cabrera High School                71.359526        90.850377   
Bailey High School                 71.008842        91.639871   
Wright High School                 70.969444        91.777778   
Rodriguez High School              70.935984        90.797699   

                       % Passing Reading  % Overall Passing  
Griffin High School            88.487738          81.335150  
Cabrera High School            89.074273          80.785791  
Bailey High School             87.379421          80.084405  
Wright High School             86.666667          79.722222  
Rodriguez High School          87.396849          79.419855

## Bottom Performing Schools (By % Overall Passing)

In [296]:
#Sort the schools by % Overall Passing in ascending order and display the top 5 rows.
#Save the results in a DataFrame called "bottom_schools".
bottom_schools_df = per_school_summary_df.sort_values(["% Overall Passing"], ascending=True)
bottom_schools_df.head(5)

School Type  Total Students   Total Budget  \
Hernandez High School   Government            4635  $3,022,020.00   
Huang High School       Government            2917  $1,910,635.00   
Johnson High School     Government            4761  $3,094,650.00   
Wilson High School     Independent            2283  $1,319,574.00   
Ford High School        Government            2739  $1,763,916.00   

                      Per Student Budget  Average Maths Score  \
Hernandez High School            $652.00            68.874865   
Huang High School                $655.00            68.935207   
Johnson High School              $650.00            68.843100   
Wilson High School               $578.00            69.170828   
Ford High School                 $644.00            69.091274   

                       Average Reading Score  % Passing Maths  \
Hernandez High School              69.186408        80.949299   
Huang High School                  68.910525        81.693521   
Johnson High School                69.039277        82.062592   
Wilson High School                 68.876916        82.785808   
Ford High School                   69.572472        82.438846   

                       % Passing Reading  % Overall Passing  
Hernandez High School          81.877023          66.364617  
Huang High School              81.453548          66.712376  
Johnson High School            81.978576          67.191766  
Wilson High School             81.296540          67.455103  
Ford High School               82.219788          67.469880

## Maths Scores by Year

In [297]:
#Perform the necessary calculations to create a DataFrame that lists the average maths score for students of each year level (9, 10, 11, 12) at each school.
#Create data series per year
Year_nine = school_student_complete_data[(school_student_complete_data["year"] == 9)]
Year_ten = school_student_complete_data[(school_student_complete_data["year"] == 10)]
Year_eleven = school_student_complete_data[(school_student_complete_data["year"] == 11)]
Year_twelve = school_student_complete_data[(school_student_complete_data["year"] == 12)]

#Group by school
Year_nine_scores = Year_nine.groupby(["school_name"])["maths_score"].mean()
Year_ten_scores = Year_ten.groupby(["school_name"])["maths_score"].mean()
Year_eleven_scores = Year_eleven.groupby(["school_name"])["maths_score"].mean()
Year_twelve_scores = Year_twelve.groupby(["school_name"])["maths_score"].mean()

# #Combine the series into a dataframe
maths_scores = pd.DataFrame({"Year9": Year_nine_scores, "Year10": Year_ten_scores, "Year11": Year_eleven_scores,
                                 "Year12": Year_twelve_scores})
maths_scores                                    

Year9     Year10     Year11     Year12
school_name                                                      
Bailey High School     72.493827  71.897498  72.374900  72.675097
Cabrera High School    72.321970  72.437768  71.008299  70.604712
Figueroa High School   68.477804  68.331586  68.811001  69.325282
Ford High School       69.021609  69.387006  69.248862  68.617811
Griffin High School    72.789731  71.093596  71.692521  71.469178
Hernandez High School  68.586831  68.867156  69.154412  68.985075
Holden High School     70.543307  75.105263  71.640777  73.409639
Huang High School      69.081754  68.533246  69.431345  68.639316
Johnson High School    69.469286  67.990220  68.637730  69.287393
Pena High School       71.996364  72.396000  72.523438  71.187845
Rodriguez High School  71.940722  71.779808  72.364811  72.154626
Shelton High School    72.932075  72.506696  70.097087  72.331536
Thomas High School     69.234273  70.057007  69.657831  69.369822
Wilson High School     69.212361  69.455446  68.378965  69.787472
Wright High School     71.741176  72.179012  73.275862  70.848238

## Reading Score by Year

In [298]:
#Create a DataFrame that lists the average reading score for students of each year level (9, 10, 11, 12) at each school.
#Create data series per year
Year_nine_reading = school_student_complete_data[(school_student_complete_data["year"] == 9)]
Year_ten_reading = school_student_complete_data[(school_student_complete_data["year"] == 10)]
Year_eleven_reading = school_student_complete_data[(school_student_complete_data["year"] == 11)]
Year_twelve_reading = school_student_complete_data[(school_student_complete_data["year"] == 12)]

#Group by school
Year_nine_reading_scores = Year_nine_reading.groupby(["school_name"])["reading_score"].mean()
Year_ten_reading_scores = Year_ten_reading.groupby(["school_name"])["reading_score"].mean()
Year_eleven_reading_scores = Year_eleven_reading.groupby(["school_name"])["reading_score"].mean()
Year_twelve_reading_scores = Year_twelve_reading.groupby(["school_name"])["reading_score"].mean()

# #Combine the series into a dataframe
reading_scores = pd.DataFrame({"Year9": Year_nine_reading_scores, "Year10": Year_ten_reading_scores, 
                                "Year11": Year_eleven_reading_scores, "Year12": Year_twelve_reading_scores})
reading_scores

Year9     Year10     Year11     Year12
school_name                                                      
Bailey High School     70.901920  70.848265  70.317346  72.195525
Cabrera High School    71.172348  71.328326  71.201245  71.856021
Figueroa High School   70.261682  67.677588  69.152327  69.082126
Ford High School       69.615846  68.988701  70.735964  68.849722
Griffin High School    72.026895  70.746305  72.385042  69.434932
Hernandez High School  68.477569  70.621842  68.418199  69.244136
Holden High School     71.598425  71.096491  73.310680  70.481928
Huang High School      68.670616  69.516297  68.740638  68.671795
Johnson High School    68.719286  69.295029  69.969115  67.992521
Pena High School       70.949091  72.324000  71.703125  71.513812
Rodriguez High School  70.902921  70.137500  71.424453  71.414449
Shelton High School    70.715094  69.879464  71.150485  69.070081
Thomas High School     69.672451  69.741093  70.749398  68.730769
Wilson High School     68.683043  68.412541  68.796327  69.888143
Wright High School     71.823529  71.386831  70.475862  69.821138

## Scores by School Spending

In [299]:
#Create a table that breaks down school performance based on average spending ranges (per student).
#Use the code provided to create four bins with reasonable cutoff values to group school spending.
spending_bins = [0, 585, 630, 645, 680]
group_labels = ["<$585", "$585-630", "$630-645", "$645-680"]

per_school_summary_df["Spending Ranges (Per Student)"] = pd.cut(per_student_budget, spending_bins, labels=group_labels)

spending_maths_scores = per_school_summary_df.groupby(["Spending Ranges (Per Student)"])["Average Maths Score"].mean()
spending_reading_scores = per_school_summary_df.groupby(["Spending Ranges (Per Student)"])["Average Reading Score"].mean()
spending_passing_maths = per_school_summary_df.groupby(["Spending Ranges (Per Student)"])["% Passing Maths"].mean()
spending_passing_reading = per_school_summary_df.groupby(["Spending Ranges (Per Student)"])["% Passing Reading"].mean()
spending_overall_passing = per_school_summary_df.groupby(["Spending Ranges (Per Student)"])["% Overall Passing"].mean()

spending_summary =pd.DataFrame({"Average Maths Score": spending_maths_scores, "Average Reading Score": spending_reading_scores,
                                    "% Passing Maths": spending_passing_maths, "% Passing Reading": spending_passing_reading,
                                    "% Overall Passing": spending_overall_passing})
spending_summary

Average Maths Score  Average Reading Score  \
Spending Ranges (Per Student)                                               
<$585                                    71.364587              70.716577   
$585-630                                 72.065868              71.031297   
$630-645                                 69.854807              69.838814   
$645-680                                 68.884391              69.045403   

                               % Passing Maths  % Passing Reading  \
Spending Ranges (Per Student)                                       
<$585                                88.835926          86.390517   
$585-630                             91.518824          87.292423   
$630-645                             84.686139          83.763585   
$645-680                             81.568470          81.769716   

                               % Overall Passing  
Spending Ranges (Per Student)                     
<$585                                  76.721458  
$585-630                               79.876293  
$630-645                               71.004977  
$645-680                               66.756253

## Scores by School Size

In [300]:
#Find school performance based on school size (small, medium and large)
size_bins = [0, 1000, 2000, 5000]
size_labels = ["Small (<1000)", "Medium (1000-2000)", "Large (2000-5000)"]

per_school_summary_df["School Size"] = pd.cut(school_students, size_bins, labels=size_labels)

size_maths_scores = per_school_summary_df.groupby(["School Size"])["Average Maths Score"].mean()
size_reading_scores = per_school_summary_df.groupby(["School Size"])["Average Reading Score"].mean()
size_passing_maths = per_school_summary_df.groupby(["School Size"])["% Passing Maths"].mean()
size_passing_reading = per_school_summary_df.groupby(["School Size"])["% Passing Reading"].mean()
size_overall_passing = per_school_summary_df.groupby(["School Size"])["% Overall Passing"].mean()

size_summary =pd.DataFrame({"Average Maths Score": size_maths_scores, "Average Reading Score": size_reading_scores,
                                    "% Passing Maths": size_passing_maths, "% Passing Reading": size_passing_reading,
                                    "% Overall Passing": size_overall_passing})
size_summary

Average Maths Score  Average Reading Score  \
School Size                                                      
Small (<1000)                 72.335748              71.636864   
Medium (1000-2000)            71.421650              70.720164   
Large (2000-5000)             69.751809              69.576052   

                    % Passing Maths  % Passing Reading  % Overall Passing  
School Size                                                                
Small (<1000)             90.806867          87.557513          79.066348  
Medium (1000-2000)        89.846560          86.714149          78.039785  
Large (2000-5000)         84.252804          83.301185          70.293507

## Scores by School Type

In [301]:
#Find school performance based on school type

type_maths_scores = per_school_summary_df.groupby(["School Type"])["Average Maths Score"].mean()
type_reading_scores = per_school_summary_df.groupby(["School Type"])["Average Reading Score"].mean()
type_passing_maths = per_school_summary_df.groupby(["School Type"])["% Passing Maths"].mean()
type_passing_reading = per_school_summary_df.groupby(["School Type"])["% Passing Reading"].mean()
type_overall_passing = per_school_summary_df.groupby(["School Type"])["% Overall Passing"].mean()

type_summary =pd.DataFrame({"Average Maths Score": type_maths_scores, "Average Reading Score": type_reading_scores,
                                    "% Passing Maths": type_passing_maths, "% Passing Reading": type_passing_reading,
                                    "% Overall Passing": type_overall_passing})
type_summary

Average Maths Score  Average Reading Score  % Passing Maths  \
School Type                                                                
Government             69.834806              69.675929        84.462375   
Independent            71.368822              70.718933        89.204043   

             % Passing Reading  % Overall Passing  
School Type                                        
Government           83.587562          70.698993  
Independent          86.247789          76.973340

## Analysis

The above data breaks down PyCity Schools into two main areas. Firstly the entire Local Government area and secondly the comparison of each school within the area, with a focus given to school budgets along with maths and reading results.
There are 15 schools within the Local Government area with a total of 39,170 students.

The following conclusions can be made from the above 
1. 3 of the top 5 performing schools (% overall passing) are Independent schools. Griffin High School ranks as the best performing school (81.3%) despite having the 3rd lowest student population (917,500).

2. 4 of the bottom 5 performing schools (% overall passing) are Government schools. Hernandez High School ranks as the worst performing school (66.3%) despite having the 3rd highest student population (3,022,020).

In summary a higher school population would seemingly result in a lower overall paasing % when combining Maths and Reading scores. When comparing Griffin and Hernandez High Schools, Hernandez spends an extra $27.00 per student (Hernandez $652, Griffin $625) which results in a 15% worse overall passing rate when combining Maths and Reading scores. (Hernandez 66.3, Griffin 81.3)

3. Both Math scores and Reading scores across all schools are relatively even through Year levels 9 - 12 ranging between high 60s(%) and low 70s(%) for both Maths and Reading.

4. Looking into the Spending Ranges (Per Student) it would seem that the higher ranges lack the results of the lower ranges. Lower ranges seem to out perform the higher ranges across Maths, Reading and Overall %. The average Maths and Reading scores are closer but the $630,000 and below ranges still show to have better results.

5. The same can be said for smaller schools (student population). These also outperform the medium and larger sized schools in all result areas.

6. As noted in points 1. and 2. Independent schools achieve higher average scores across all selected fields.



*** This analysis has also been saved to a seperate txt file.
